ibis/backends/base/sql/alchemy/client.py

```diff
@@ -329,3 +329,120 @@ def _sqla_table_to_expr(self, table):
     def version(self):
         vstring = '.'.join(map(str, self.con.dialect.server_version_info))
         return parse_version(vstring)
+
+    def insert(
+        self,
+        to_table_name: str,
+        database: Optional[str] = None,
+        data_obj=None,
+        from_table_name: Optional[str] = None,
+        if_exists: Optional[str] = 'append',
+    ) -> None:
+        """
+        Insert the given data to a table in backend.
+
+        Parameters
+        ----------
+        to_table_name : string
+            name of the table to which data needs to be inserted
+        database : string, optional
+            name of the attached database that the table is located in.
+        data_obj : pd.Dataframe
+            data_obj is the dataframe containing data which needs to be
+            inserted to to_table_name
+        from_table_name : string, optional
```

Comment:
umm, this is really strange to have 2 tables here?

Reply:
As mentioned in the previous comment, `from_table_name` is being used when `data` (dataframe) is not sent directly but when we want to insert data (all the data) from another table in the database.

Reply:
I thought you copied this signature from Impala, but I see you created your own.

I see that in Spark and Impala we've got the next signatures:
```
backends/spark/client.py:    def insert(
backends/spark/client.py-        self,
backends/spark/client.py-        table_name,
backends/spark/client.py-        obj=None,
backends/spark/client.py-        database=None,
backends/spark/client.py-        overwrite=False,
backends/spark/client.py-        values=None,
backends/spark/client.py-        validate=True,
backends/spark/client.py-    ):

backends/impala/client.py:    def insert(
backends/impala/client.py-        self,
backends/impala/client.py-        table_name,
backends/impala/client.py-        obj=None,
backends/impala/client.py-        database=None,
backends/impala/client.py-        overwrite=False,
backends/impala/client.py-        partition=None,
backends/impala/client.py-        values=None,
backends/impala/client.py-        validate=True,
backends/impala/client.py-    ):
```

Where did you get the signature you implemented from?

Happy to keep `if_exists` since you already implemented it. And I think we can ignore `values` for now, I think we should get rid of it, and pass values in `obj` instead. But for the rest, please be consistent with the signature of the other functions. Ideally users can write code for one backend, and unless there is a good reason, should work in all others.

Reply:
I was more or less trying to use the signature from impala itself.  
After a couple of requested changes, I did drop the `values` parameter and its usage. I've also updated the `data_obj` parameter to `obj` like in impala.  
I couldn't use `table_name` as is because we added a separate workflow if a table was passed (`from_table_name`) for inserting data into a table (`to_table_name`). If it's fine, shall I update the `to_table_name` parameter to `table_name`?  
Let me check how `overwrite` is being used in impala and update `if_exists` to it. Since `overwrite` is a boolean and `if_exists` is not (it's a string) do you want me to make `overwrite` a string parameter or update the code to use it as a boolean?

Reply:
Updated `to_table_name` variable to `table_name`.

Reply:
`if_exists` is also being used in the `load_data` function of the SQLAlchemy backend but not `overwrite`, though `overwrite` is being used in Impala. So, that was the reason behind my implementation having `if_exists` and not `overwrite`.  
It's not a big task to modify the functionality and test cases. Please let me know what we want to go forward with @datapythonista.
Thanks!

Reply:
So, the idea would be that if an Ibis user wrote a call to `insert`  in Impala, and changes backend to postgres, the code should still work. This is not always 100% possible, for example, the `partition` parameter is specific to Impala. But for the rest, would be good to keep all `insert` methods consistent.

So, would be good to change:
- Merge `data` and `from_table_name` into a single `obj` parameter, and then check the type to see which operation has to be done.
- Replace `if_exists` by `overwrite` as a boolean. We will lose the `fail` case, but the users can easily check themselves `if con.table('whatever').exists():` or similar to implement that.
- Change the order of parameters so it matches the order of the Impala/Spark `insert`
- Any other change that makes sense

I think it should be quite quick to change this now that you've got your implementation working. Thanks for all the hard work in this PR, and apologies for not realizing on this earlier, and incorrectly assumed that you were using the other `insert` methods as a reference.

Reply:
Thanks for the inputs @datapythonista.  
It was my mistake as well to not implement the functionality similar to impala ones and check with you.  
Anyways, I've updated the `insert` function in SQLAlchemy, and the test functions with the above suggestions. Please take a look and let me know if there's anything I'm still missing.
Thanks once again!